backend/server-connection/src/test/java/org/sonarsource/sonarlint/core/serverconnection/ServerInfoSynchronizerTests.java

```diff
@@ -68,14 +70,19 @@ void it_should_read_version_from_storage_when_available() throws IOException {
   }
 
   @Test
-  void it_should_synchronize_version_when_not_available() {
+  void it_should_synchronize_version_and_mode_when_not_available() {
     mockServer.addStringResponse("/api/system/status", "{\"id\": \"20160308094653\",\"version\": \"9.9\",\"status\": \"UP\"}");
+    mockServer.addProtobufResponse("/api/settings/values.protobuf?keys=sonar.multi-quality-mode.enabled", Settings.ValuesWsResponse.newBuilder()
+      .addSettings(Settings.Setting.newBuilder()
+        .setKey("sonar.multi-quality-mode.enabled")
+        .setValue("true"))
+      .build());
 
     var storedServerInfo = synchronizer.readOrSynchronizeServerInfo(new ServerApi(mockServer.endpointParams(), HttpClientProvider.forTesting().getHttpClient()), new SonarLintCancelMonitor());
 
     assertThat(storedServerInfo)
-      .extracting("version")
-      .hasToString("9.9");
+      .extracting("version", "isMQRMode")
```

Comment:
Nitpick: you could use getters references here (e.g. `StoredServerInfo::getVersion`), this works better if a renaming occurs